pkg/function-proxy/utils/proxy-utils.go

```diff
@@ -36,21 +37,26 @@ var (
 	shutdownTimeout    = os.Getenv("SHUTDOWN_TIMEOUT")
 	intTimeout         int
 	intShutdownTimeout int
-	funcHistogram      = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	FuncHistogram      = promauto.NewHistogramVec(prometheus.HistogramOpts{
 		Name: "function_duration_seconds",
 		Help: "Duration of user function in seconds",
 	}, []string{"method"})
-	funcCalls = prometheus.NewCounterVec(prometheus.CounterOpts{
+	FuncCalls = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "function_calls_total",
 		Help: "Number of calls to user function",
 	}, []string{"method"})
-	funcErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+	FuncErrors = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name: "function_failures_total",
 		Help: "Number of exceptions in user function",
 	}, []string{"method"})
 )
 
+func PromHTTPHandler() http.Handler{
+	return promhttp.Handler()
+}
```

Comment:
Correct me if I am wrong but I think this is the only needed change. You can still use prometheus instead than promauto and avoid exposing `FuncHistogram`, `FuncCalls` and `FuncErrors` so the only change in this PR would be exposing this function. Can you try that?

Reply:
well I opted for `promauto` anyway because if we are to add more metrics in future we wont be needed/bothered to register them all, wherever in whichever function we define them, they will be registered automatically. It makes this metrics part a bit easier. 
And `FuncHistogram` instead of `funcHistogram` for if we want to increment them in some other packages in future. 
Thoughts?

Reply:
but those are different topics, the goal of this PR is to fix the current metrics so I would leave it as minimal as possible to avoid confusion in the future of what did the trick. Then we can open a different PR to modify/add more metrics if needed.

Reply:
sure, we can take this up later.